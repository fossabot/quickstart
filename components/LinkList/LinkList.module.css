.LinkList {
  display: grid;
  grid-template-rows: 1fr 1fr;
  grid-template-columns: 1fr;
  row-gap: 10px;
}

@media (min-width: 620px) {
  .LinkList {
    grid-template-columns: 1fr 1fr;
    column-gap: 20px;
    row-gap: 20px;
  }
}

@media all and (-ms-high-contrast: none) {
  .LinkList {
    display: -ms-grid;
    -ms-grid-columns: 1fr 1fr;
    -ms-grid-rows: 1fr 1fr;
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .LinkItem {
    display: flex;
    align-items: flex-start;
    padding: 20px 30px 20px 15px;
    color: #333;
    border: 1px solid #00000010;
    border-radius: 5px;
    transition: none;
  }
}

.LinkItem {
  display: flex;
  align-items: flex-start;
  padding: 20px 30px 20px 15px;
  color: #333;
  border: 1px solid #00000010;
  border-radius: 5px;
  transition: background 150ms ease-out, border 150ms ease-out,
    transform 150ms ease-out;
}

.LinkItem:hover,
.LinkItem:focus {
  border-color: #1ea7fd50;
  box-shadow: rgba(0, 0, 0, 0.08) 0 3px 10px 0;
  transform: translate3d(0, -3px, 0);
}

.LinkItem:active {
  border-color: #1ea7fd;
  transform: translate3d(0, 0, 0);
}

.LinkItem strong {
  display: block;
  margin-bottom: 2px;
  font-weight: 700;
}

.LinkItem svg {
  flex: none;
  width: 40px;
  height: 40px;
  margin-right: 15px;
}

.LinkItem span {
  font-size: 14px;
  line-height: 20px;
}
