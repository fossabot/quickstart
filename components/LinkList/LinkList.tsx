import React from 'react'

import ErrorBoundary from '../../util/ErrorBoundary'
import Styles from './LinkList.module.css'

export interface LinkListItem {
  /** The url of the link */
  href: string
  /** The title of the link */
  title: string
  /** The text description of the link */
  description: string
}

export interface LinkListProps {
  /** The collection of links to display */
  links: LinkListItem[]
}

export function LinkList({ links }: LinkListProps): JSX.Element {
  return (
    <ErrorBoundary>
      <div className={Styles.LinkList}>
        {links.map(({ href, title, description }) => (
          <a className={Styles.LinkItem} href={href} key={title}>
            <span>
              <strong>{title}</strong>
              {description}
            </span>
          </a>
        ))}
      </div>
    </ErrorBoundary>
  )
}

export default LinkList
