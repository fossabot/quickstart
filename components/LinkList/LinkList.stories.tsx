import { Meta, Story } from '@storybook/react'
import React from 'react'

import { LinkList } from './LinkList'

const config: Meta = {
  title: 'Components/LinkList',
  component: LinkList,
}

export default config

const Template: Story = (args: any) => <LinkList {...args} />

export const Default = Template.bind({})
Default.args = {
  links: [
    {
      href: 'https://nextjs.org/docs/routing/introduction',
      title: 'File system routing with Next.js',
      description: "Learn the basics of Next.js's file-based routing system",
    },
    {
      href: 'https://nextjs.org/docs/api-routes/introduction',
      title: 'Handle API requests with Next.js',
      description: 'Learn how to create a simple API in Next.js',
    },
    {
      href:
        'https://github.com/vinissimus/next-translate#use-translations-in-your-pages',
      title: 'Adding text localisations with Next-Translate',
      description: 'Quickly set up translation keys for your page or component',
    },
    {
      href: 'https://github.com/garmeeh/next-seo#add-seo-to-page',
      title: 'Adding SEO Metadata with NextSEO',
      description:
        'Quickly add necessary SEO information to a page using next-seo',
    },
    {
      href: 'https://next-auth.js.org/getting-started/client#usesession',
      title: 'Handling authentication with Next-Auth',
      description:
        'Check if the user is logged in or not and change the behaviour of a page',
    },
    {
      href: 'https://www.youtube.com/watch?v=lgurVvQsOTY',
      title: 'End-To-End (E2E) Testing with Cypress',
      description:
        'Gleb Bahmutov talks about how you can use Cypress.io, a free open-source testing tool, to write end-to-end tests for your React apps.',
    },
    {
      href:
        'https://github.com/cyrilwanner/next-optimized-images/tree/canary#image-components',
      title: 'Including images with Next-Optimized-Images',
      description:
        'Have your assets automatically optimized using the amazing next-optimized-images plugin',
    },
  ],
}

Default.parameters = {
  jest: ['LinkList.test.tsx'],
}
