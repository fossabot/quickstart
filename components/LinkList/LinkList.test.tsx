import React from 'react'

import { LinkList } from './LinkList'

export const title = 'Unit/Components/LinkList'

const LinkListProps = {
  links: [
    {
      href: 'https://nextjs.org/docs/routing/introduction',
      title: 'File system routing with Next.js',
      description: "Learn the basics of Next.js's file-based routing system",
    },
  ],
}
describe(title, () => {
  it('Should render without an error', () =>
    expect(() => <LinkList {...LinkListProps} />).not.toThrow())

  it('Should match the snapshot', () =>
    expect(<LinkList {...LinkListProps} />).toMatchSnapshot())
})
