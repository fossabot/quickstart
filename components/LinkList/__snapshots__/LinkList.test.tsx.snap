// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit/Components/LinkList Should match the snapshot 1`] = `
<LinkList
  links={
    Array [
      Object {
        "description": "Learn the basics of Next.js's file-based routing system",
        "href": "https://nextjs.org/docs/routing/introduction",
        "title": "File system routing with Next.js",
      },
    ]
  }
/>
`;
